fs/ftpfs.py

```diff
@@ -720,6 +720,11 @@ def _scandir(
                         raise  # pragma: no cover
                 if lines:
                     for raw_info in self._parse_mlsx(lines):
+                        try:
```

Comment:
I think in principle this is the fix. But it may be best done in `_parse_mlsx`.

I'm thinking that it should just skip over anything that is not a `file` or `dir`.